pkg/controller/controller.go

```diff
@@ -353,22 +367,27 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
 		}
 	}
 
-	var needSnapshotSupport bool
+	// Make sure the plugin is capable of fulfilling the requested options
+	rc := &requiredCapabilities{}
 	if options.PVC.Spec.DataSource != nil {
 		// PVC.Spec.DataSource.Name is the name of the VolumeSnapshot API object
 		if options.PVC.Spec.DataSource.Name == "" {
 			return nil, fmt.Errorf("the PVC source not found for PVC %s", options.PVC.Name)
 		}
-		if options.PVC.Spec.DataSource.Kind != snapshotKind {
-			return nil, fmt.Errorf("the PVC source is not the right type. Expected %s, Got %s", snapshotKind, options.PVC.Spec.DataSource.Kind)
-		}
-		if *(options.PVC.Spec.DataSource.APIGroup) != snapshotAPIGroup {
-			return nil, fmt.Errorf("the PVC source does not belong to the right APIGroup. Expected %s, Got %s", snapshotAPIGroup, *(options.PVC.Spec.DataSource.APIGroup))
+
+		switch options.PVC.Spec.DataSource.Kind {
```

Comment:
@j-griffith  we already have a `getVolumeContentSource`, can we make use of it?

Reply:
Hi @humblec thanks for pointing that out.  Actually you'll notice I *do* utilize that helper function, I added the fetch for PVC to it and call it when we actually pull the content.

This particular check in this section was a here to validate the options and "fail fast" if they were invalid, as well as set a variable so that the provision stage later would have a flag to "getContentSource" or not.  If we want to consolidate things into a single helper, that's certainly possible, but it seemed like following the flow we already have for Snapshots was a good idea here.